File: win-prog/ProgWinCS/Taming the Mouse/CheckerWithChildren/CheckerWithChildren.cs
//--------------------------------------------------
// CheckerWithChildren.cs © 2001 by Charles Petzold
//--------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class CheckerWithChildren: Form
{
     protected const int       xNum = 5;
     protected const int       yNum = 4;
     protected CheckerChild[,] acntlChild;

     public static void Main()
     {
          Application.Run(new CheckerWithChildren());
     }
     public CheckerWithChildren()
     {
          Text = "Checker with Children";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;
          CreateChildren();

		 OnResize(EventArgs.Empty);
     }
     protected virtual void CreateChildren()
     {
          acntlChild = new CheckerChild[yNum, xNum];

          for (int y = 0; y < yNum; y++)
          for (int x = 0; x < xNum; x++)
          {
               acntlChild[y, x] = new CheckerChild();
               acntlChild[y, x].Parent = this;
          }
     }
     protected override void OnResize(EventArgs ea)
     {
          int cxBlock = ClientSize.Width / xNum;
          int cyBlock = ClientSize.Height / yNum;

          for (int y = 0; y < yNum; y++)
          for (int x = 0; x < xNum; x++)
          {
               acntlChild[y, x].Location = new Point(x*cxBlock, y*cyBlock);
               acntlChild[y, x].Size     = new Size(cxBlock, cyBlock);
          }
     }
}

File: win-prog/ProgWinVB/Taming the Mouse/CheckerWithChildren/CheckerWithChildren.vb
'----------------------------------------------------
' CheckerWithChildren.vb (c) 2002 by Charles Petzold
'----------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class CheckerWithChildren
    Inherits Form

    Protected Const xNum As Integer = 5
    Protected Const yNum As Integer = 4
    Protected actrlChild(,) As CheckerChild

    Shared Sub Main()
        Application.Run(New CheckerWithChildren())
    End Sub

    Sub New()
        Text = "Checker with Children"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText
        CreateChildren()
        OnResize(EventArgs.Empty)
    End Sub

    Protected Overridable Sub CreateChildren()
        ReDim actrlChild(yNum - 1, xNum - 1)
        Dim x, y As Integer

        For y = 0 To yNum - 1
            For x = 0 To xNum - 1
                actrlChild(y, x) = New CheckerChild()
                actrlChild(y, x).Parent = Me
            Next x
        Next y
    End Sub

    Protected Overrides Sub OnResize(ByVal ea As EventArgs)
        Dim cxBlock As Integer = ClientSize.Width \ xNum
        Dim cyBlock As Integer = ClientSize.Height \ yNum
        Dim x, y As Integer

        For y = 0 To yNum - 1
            For x = 0 To xNum - 1
                actrlChild(y, x).Location = _
                            New Point(x * cxBlock, y * cyBlock)
                actrlChild(y, x).Size = New Size(cxBlock, cyBlock)
            Next x
        Next y
    End Sub
End Class
